### PR TITLE
Validate https default port for custom nodes

### PR DESCRIPTION
**Pull request by laumair**
# Description

There is no validation for https port for custom nodes. Example:
Consider a node (https://pow1.iota.community:443) which is already added to the list of nodes in Trinity. A user can add https://pow1.iota.community and Trinity will allow this addition.

This behavior can raise problems in nodes selection within quorum where two nodes pointing to the same domain and port are selected.

## Type of change

- Enhancement

# How Has This Been Tested?

- Manually tested on iOS, android and desktop (debug) mode

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

**Review by cvarley100 — CHANGES_REQUESTED** [review]
It looks like the mobile code is duplicated in the shared HOC. I suggest using the HOC for both desktop and mobile.

**Comment by laumair** [comment]
@cvarley100 Will submit a separate PR for using `Node` HOC in mobile.

File: src/mobile/src/ui/views/wallet/AddCustomNode.js [hunk 1]
```diff
@@ -193,7 +195,16 @@ class AddCustomNode extends Component {
...
-        if (!nodes.includes(customNode.replace(/ /g, ''))) {
+        const hasDefaultHttpsPort = endsWith(customNode, ':443');
+
+        if (hasDefaultHttpsPort) {
+            customNode = customNode.slice(0, -4);
+        }
+
+        if (
+            !nodes.includes(customNode) &&
+            !map(nodes, (node) => (endsWith(node, ':443') ? node.slice(0, -4) : node)).includes(customNode)
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/AddCustomNode.js]: Why do we use `lodash.map` here but `Array.prototype.map` at https://github.com/iotaledger/trinity-wallet/pull/849/files#diff-14d3a021ba31141b14312a6343d8a557R65?

**Comment by laumair** [comment]
@cvarley100 Done [here](https://github.com/iotaledger/trinity-wallet/pull/853)

File: src/mobile/src/ui/views/wallet/AddCustomNode.js (hunk 1 above)
Comment (laumair) [reply src/mobile/src/ui/views/wallet/AddCustomNode.js]: No specific reason. Both are iterating over arrays here. 

However, https://github.com/iotaledger/trinity-wallet/pull/853 will remove this duplication.

Comment (rajivshah3) [reply src/mobile/src/ui/views/wallet/AddCustomNode.js]: Oh ok. I was just wondering if there was a special reason

**Review by cvarley100 — APPROVED** [review]

**Pull request merged**